lib/src/scanner/mod.rs

```diff
@@ -457,6 +499,26 @@ impl<'r> Scanner<'r> {
         Ok(())
     }
 
+    /// Updates the metadata for a module specified by its fully-qualified name.
+    ///
+    /// If the `meta` argument is `None`, the metadata for the module is removed.
+    ///
+    /// See [`Scanner::module_meta`] for the reasoning behind choosing `Arc<_>`
+    pub fn set_module_meta(
+        &mut self,
+        module_full_name: &str,
```

Comment:
For simplicity this API should receive the module name ("pe", "elf", "macho", etc). The concept of fully-qualified name an internal concept, and most API users won't know what is this. It's ok if the API can work with both names, but if only one of them is accepted, it should be short name.

Reply:
good catch, the name remained after a version of the code that used the full module name to identify the modules. this was later changed, the name was not

fixed in [cca5adf](https://github.com/VirusTotal/yara-x/pull/183/commits/cca5adfaeb3671482a47f5b0a1a8e3f7d2a70fe8)